source/auth/tests/connection-string.yml

```diff
@@ -291,3 +291,11 @@ tests:
         description: "authSource without username is invalid (default mechanism)"
         uri: "mongodb://localhost/?authSource=foo"
         valid: false
+
+        description: "should throw an exception if no username provided (userinfo implies default mechanism)"
+        uri: "mongodb://@localhost.com/"
+        valid: false
+
```

Comment:
@haris-sheikh you missed a `-` here which means running `make` in the top-level `source/` directory no longer works: 
```
$ make
tmpfile=$(mktemp);               \
	if js-yaml auth/tests/connection-string.yml > "$tmpfile"; then \
	    mv "$tmpfile" auth/tests/connection-string.json;            \
	else                              \
	    rm "$tmpfile";               \
	    exit 1;                       \
	fi
YAMLException: duplicated mapping key at line 295, column 9:
            description: "should throw an ex ...
            ^
make: *** [auth/tests/connection-string.json] Error 1
```

Reply:
@haris-sheikh: In the future, each PR with YAML changes should regenerate JSON files (see: https://github.com/mongodb/specifications#converting-to-json). That would have caught this mistake before merging, so please keep in mind for any future PRs.

Apart from that, all of the reviewers (myself included) missed that this PR was changing YAML without including a corresponding JSON change.

Reply:
Oh okay, sorry for that. Will keep this in mind for any future PRs!